modules/monitoring-resolving-the-kubepersistentvolumefillingup-alert-firing-for-prometheus.adoc

```diff
@@ -0,0 +1,68 @@
+// Module included in the following assemblies:
+//
+// * monitoring/troubleshooting-monitoring-issues.adoc
+// * support/troubleshooting/investigating-monitoring-issues.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="resolving-the-kubepersistentvolumefillingup-alert-firing-for-prometheus_{context}"]
+= Resolving the KubePersistentVolumeFillingUp alert firing for Prometheus
+
+As a cluster administrator, you can resolve the `KubePersistentVolumeFillingUp` alert being triggered for Prometheus. 
+
+The critical alert fires when a persistent volume (PV) claimed by a `prometheus-k8s-*` pod in the `openshift-monitoring` project has less than 3% of its total space remaining. This can cause Prometheus to not function normally.
+
+[NOTE]
+====
+There are two `KubePersistentVolumeFillingUp` alerts:
+
+* *Critical alert*: It is triggered when the mounted PV has less than 3% of its total space remaining. 
+* *Warning alert*: It is triggered when the mounted PV has less than 15% of its total space remaining and is expected to fill up within four days.
+====
+
+You can resolve this issue by freeing some space for the PV by removing the three oldest Prometheus time series database (TSDB) blocks.
+
+.Prerequisites
+
+ifndef::openshift-dedicated,openshift-rosa[]
+* You have access to the cluster as a user with the `cluster-admin` cluster role.
+endif::openshift-dedicated,openshift-rosa[]
+ifdef::openshift-dedicated,openshift-rosa[]
+* You have access to the cluster as a user with the `dedicated-admin` role.
+endif::openshift-dedicated,openshift-rosa[]
+* You have installed the OpenShift CLI (`oc`).
+
+.Procedure
+
+. Remove the three oldest Prometheus TSDB blocks to make more space available for the PV:
++
+[source,terminal]
+----
+$ oc debug <prometheus_k8s_pod_name> -n openshift-monitoring \// <1>
+-c prometheus --image=$(oc get po <prometheus_k8s_pod_name> \// <1>
+-o jsonpath='{.spec.containers[0].image}') \
+-- sh -c 'ls -latr /prometheus/ | egrep -o "[0-9|A-Z]{26}" | head -3 | \
```

Comment:
is the `egrep -o "[0-9|A-Z]{26}"` right, seems it's wrong, need the dev for confirm

Reply:
no issue now
```
$ oc -n openshift-monitoring rsh -c prometheus prometheus-k8s-0
sh-5.1$ ls /prometheus
01HTYTFRZHJAZ8WPP1XZTTFHB0  chunks_head  lock  queries.active  wal
```
`01HTYTFRZHJAZ8WPP1XZTTFHB0` matches `egrep -o "[0-9|A-Z]{26}"`
https://prometheus.io/docs/prometheus/latest/storage/#on-disk-layout also prove it